packages/nimble-components/src/rich-text/editor/models/create-tiptap-editor.ts

```diff
@@ -194,13 +200,31 @@ function createCustomMentionExtension(
             pluginKey: new PluginKey(config.key),
             allowSpaces: true,
             render: () => {
+                let inSuggestionMode = false;
                 return {
                     onStart: (props): void => {
+                        inSuggestionMode = true;
                         config.mentionUpdateEmitter(props.query);
+                        richTextEditor.activeMentionCharacter = props.text.slice(0, 1);
+                        mentionListBox?.onMention(props);
                     },
-
                     onUpdate: (props): void => {
+                        if (!inSuggestionMode) {
+                            return;
+                        }
                         config.mentionUpdateEmitter(props.query);
+                        mentionListBox?.onMention(props);
+                    },
+                    onKeyDown: (props): boolean => {
+                        if (props.event.key === keyEscape) {
+                            inSuggestionMode = false;
+                        }
+                        return (
+                            mentionListBox?.keydownHandler(props.event) ?? false
+                        );
+                    },
+                    onExit: (): void => {
+                        mentionListBox?.close();
```

Comment:
The state of `activeMentionCharacter` should always be accurate in the editor. Seems like we need to reset it onExit right?